pkg/config/model/viper.go

```diff
@@ -96,6 +96,21 @@ func (s Source) IsGreaterOrEqualThan(x Source) bool {
 	return sourcesPriority[s] >= sourcesPriority[x]
 }
 
+// PreviousSource returns the source before the current one, or Unknown if there isn't one
+func (s Source) PreviousSource() Source {
+	if s == SourceDefault || s == SourceUnknown {
+		return SourceUnknown
+	} else if s == SourceFile {
+		return SourceDefault
```

Comment:
Should we fix the priority order instead to have unknown last ?

Reply:
Sure thing, wasn't sure if there was a reason they needed to be in this order where `unknown` == 1. It's a bit tricky so I left this change as its own commit.

Reply:
I'm undoing this change. Unfortunately, it's not possible to reorder the layers currently, because that breaks `SetWithoutSource`, which is used all over the codebase, mainly in tests.